### ISSUE TITLE
add option to sync only on WiFi

### ISSUE THREAD

**Issue by Strubbl** [wallabag/android-app#346]
With all the automatic sync options (and the image sync coming in future), i would love to see an option in the app to synchronize only on WiFi connection and not on mobile data plan.

1. case: maybe a general option to only ever sync on wifi
2. case: maybe an option to sync on wifi only if the automatic syncing is active
3. case: maybe an option to sync on wifi only if the image download is active

Which one do you think is best? Or should we go for all three (which would add a lot of complexity for sure)?

**Comment by Strubbl** [comment]
android docu for this topic https://developer.android.com/training/monitoring-device-state/connectivity-monitoring.html

**Comment by Eisfreak7** [comment]
Different usecase: I'd like to be able to sync only on a specific wifi (since the wallabag instance runs locally).
One solution would be to remove [this check](https://github.com/wallabag/android-app/blob/12e87460ba62ba646944841cf7a0b57b5b47fe10/app/src/main/java/fr/gaulupeau/apps/Poche/events/EventProcessor.java#L85), which should (I hope) allow sync to be triggered by Tasker or similar apps.
That would be very flexible and easy to implement, although maybe annoying for users that don't already have Tasker or something similar installed.

**Comment by di72nn** [comment]
@Eisfreak7 does any other app implement something like that? I wonder if there is a good way to do it.

**Comment by Eisfreak7** [comment]
Yes, [taskwarrior-android](https://bitbucket.org/kvorobyev/taskwarriorandroid/issues/6/sync-only-in-specific-network).

**Comment by Fenisu** [comment]
What about a Sync Adapter? Wouldn't that be the optimal goal?

A "system account" for wallabag, and then let Android's Doze to manage the background sync schedules, to avoid unscheduled sync wakelocks.

**Comment by Strubbl** [comment]
@Fenisu sounds very cool. Can you support here with a pull request?

**Comment by di72nn** [comment]
I considered Sync Adapter when I was implementing auto sync, but for some reason I decided not to use it. I don't remember why, so I'll probably need to look into it again.

**Comment by Fenisu** [comment]
It might be interesting to implement it if @di72nn does not find any reason to not do it.

**Comment by di72nn** [comment]
@Fenisu can you compile a list of advantages a Sync Adapter provides?

**Comment by Fenisu** [comment]
@di72nn I'm no expert but the idea would be to avoid unnecessary "independent" wakelocks, so Android and Doze can schedule the syncs when it is better suited. From there, you can also set the network constrains you want, like sync images only from wifi. And to avoid the need of GCM to monitor changes in the server, the sync can be set up to run in intervals.

The main disadvantage is that it uses by default both account authenticator and content resolver, but it can be created as dummies to ignore them and skip changing more parts of the application.

Although it might be a good idea to move the user credentials to the corresponding part of the system instead of saving them in the shared preferences xml file.

**Comment by di72nn** [comment]
The app doesn't use any explicit wakelocks (also no implicit, as far as I know). To schedule sync tasks it uses "inexact repeating" of `AlarmManager`, so Android system have some freedom in deciding when to run the sync tasks. It doesn't take network constraints into consideration though.

I don't plan reimplementing sync with Sync Adapter myself (at least anytime soon), but you may give it a shot.
